directembedding/src/main/scala/ch/epfl/directembedding/transformers/ReifyAsEmbedding.scala

```diff
@@ -63,43 +63,73 @@ trait ReifyAsEmbedding extends DirectEmbeddingModule with DirectEmbeddingUtils {
       case field @ Apply(x, y) =>
         uncurry(x, y ::: args)
       case _ =>
-        Apply(t, args.map(transform(_)))
+        Apply(t, args)
+    }
+
+    var indent: Int = 0
+    var transformDebugLevel: Int = 3
+
+    def log[T](e: T, level: Int = 0) = if (debugLevel > level) {
+      print(" " * indent)
+      self.log(s"e", level)
```

Comment:
Any better suggestion for an implementation? The indented log statements have been very helpful.